data/blog/improving-developer-experience-from-unreadable-css-selectors-to-clear-component-names.md

```diff
@@ -0,0 +1,66 @@
+---
+title: 'Improving Developer Experience: From Unreadable CSS Selectors to Clear Component Names'
+date: '2024-05-30'
+tags: ['web', 'react']
+draft: false
+summary: 'How to transform unreadable CSS selectors to React component names.'
+images: []
+layout: PostLayout
+authors: ['catherinelee']
+---
+
+Reading our own code is much easier than the output of a bundler or transpiler. The same goes for React developers reading CSS selectors generated from their CSS library of choice. CSS selectors are used all across Sentry to show clicks on components. This can provide useful information like which component is the cause of rage and dead clicks, and what component was clicked in a span. However, if you’re using React, these selectors become unreadable in production due to the minification process, so your selectors end up looking like this:
+
+`button.en302zp1.app-191aavw.e16hd6vm2[role="button"][data-test-id="common-options"]`
+
+This makes it hard to search for specific clicks or spans, and to find that component in the codebase when using search. [We made this experience better](https://sentry.io/changelog/react-component-names-is-now-available/) by utilizing component names instead of CSS selectors, so you can easily view and search for components with one simple name, like this: `CommonOptions`. In order to do this, we first had to build on our [bundler plugin](https://www.npmjs.com/package/@sentry/bundler-plugin-core) to annotate output DOM with their component name when possible.
+
+## Building the bundler plugin
+
+There are three main parts that make up the system of the component name feature: The Babel plugin, the bundler plugins, and the browser SDK. The component name plugin itself is a Babel plugin; it hooks into the transform step of your bundler’s build process, in order to attach React component names as additional properties on your compiled HTML. It parses the AST (Abstract Syntax Tree) of your JSX files at build-time to determine the names of your components, and attaches them as additional properties to the compiled HTML.
```

Comment:
We're missing some links here as there's a lot of concepts and code:

```suggestion
There are three main parts that make up the system of the component name feature: The [Babel plugin](https://github.com/getsentry/sentry-javascript-bundler-plugins/tree/main/packages/babel-plugin-component-annotate), the [bundler plugins](https://github.com/getsentry/sentry-javascript-bundler-plugins/tree/main/packages/bundler-plugin-core), and the [Sentry React SDK](https://github.com/getsentry/sentry-javascript/tree/develop/packages/react). The component name plugin itself is a Babel plugin; it hooks into the transform step of your bundler’s build process, in order to attach React component names as additional properties on your compiled HTML. It parses the AST (Abstract Syntax Tree) of your JSX files at build-time to determine the names of your components, and attaches them as additional properties to the compiled HTML.
```
Also is it in the core browser SDK or React? I assumed React so suggested that ^

Reply:
It's in the core browser SDK